### ISSUE TITLE
Module Boilerplate

### ISSUE THREAD

**Issue by ghost** [KodiCMS/kodicms-laravel#75]
Думаю что нужен такой, сразу с аннотациями. Что бы человек просто взял и стал разрабатывать.

**Comment by butschster** [comment]
Это типа пустышки?

**Comment by ghost** [comment]
Ну да - готовая иерархия файлов по папкам - сразу со всеми возможностями модуля. Можно даже с подготовленными методами в классах. Но пустые - без всякой логики. Бойлер плейтом считается такая иерархия, которая добавляется в динамическую структуру - видится системой опознается загрузчиками - но нифига не делает. Которую можно включить выключить в нашем случае.

**Comment by ghost** [comment]
Если я не прав - поправьте меня)

**Comment by butschster** [comment]
Была такая идея вообще. Пока только не понятно что нужно в него запихнуть. Я структуру плагина (модуля) описывали в wiki https://github.com/KodiCMS/kodicms-laravel/wiki/%D0%9C%D0%BE%D0%B4%D1%83%D0%BB%D0%B8 , https://github.com/KodiCMS/kodicms-laravel/wiki/%D0%9F%D0%BB%D0%B0%D0%B3%D0%B8%D0%BD%D1%8B 

В старой версии делал несколько таких плагинов, для разных целей. Для создания виджета для рабочего стола, Для создания виджета для фронта, Для раздела Datasource

**Comment by ghost** [comment]
Да нет же просто Бблаблаплагин наследуетБлаблаплагинЕкстенд имплементирует Блаблаинтерфейс
что бы такими данными был забит.

По вики все вроде понятно - но человеку писать все это нужно в ручную - либо копировать какой то модуль и очищать его. Это время. ему нужно его экономить. И если вы можете в этом помочь - то это очень здорово.

**Comment by ghost** [comment]
Ну так что - сделаем?

**Comment by butschster** [comment]
да, можно.

**Comment by ghost** [comment]
было бы здорово в ближайшее время ибо переезжать на вас очень хочется - а наработок никаких и опыта маловато. Так хоть разбирался бы потихоньку.

**Comment by butschster** [comment]
Ты лучше опиши что ты хочешь? Потому как плагин и модуль по сути одно и тоже, разница только в наличии https://github.com/KodiCMS/kodicms-laravel/blob/dev/plugins/butschster/News/PluginContainer.php

**Comment by ghost** [comment]
лучше модуль. плагин если только этим отличается - то додумаю сам уже.

**Comment by ghost** [comment]
так что насчет бойлерплейта? - может хоть подскажешь какой модуль лучше всего сейчас взять и переделать под себя?

**Comment by butschster** [comment]
Ну смотри, я бы взял плагин News, т.к. он не очень сложный и содержит в себе контроллер, модель и миграции

**Comment by ghost** [comment]
А бехейворы и эвенты туда присандалить можно?

22 октября 2015 г., 23:24 пользователь butschster notifications@github.com
написал:

> Ну смотри, я бы взял плагин News, т.к. он не очень сложный и содержит в
> себе контроллер, модель и миграции
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/KodiCMS/kodicms-laravel/issues/75#issuecomment-150346208
> .

## 

С уважением, Габриелян Д.М.

ICQ:      _3092095_
Skype:  _gt_fios_

**Comment by butschster** [comment]
Конечно, посмотри документацию по модулю там вся структура расписана. Если что задавай конкретные вопросы, я скажу что куда

**Comment by Dualse** [comment]
Поддерживаю. Можно еще сделать, чтобы через artisan создавать новый модуль с дефолтной структорой. `php artisan cms:make:module News` и развернулся новый модуль, также и с созданием новых контроллеров, миграций, сидов и т.п.
